src/libDirectoryService/ViewChangePreProcessing.cpp

```diff
@@ -95,15 +95,70 @@ bool DirectoryService::ViewChangeValidator(
     return false;
   }
 
-  if (m_mediator.m_DSCommittee->at(m_viewChangeCounter).second !=
+  // Verify candidate leader index
+  uint32_t candidateLeaderIndex = CalculateNewLeaderIndex();
```

Comment:
The other part where `CalculateNewLeaderIndex` is called doesn't seem to have the same verification code block below. Do we need to add it there? Actually here it seems unnecessary as this if condition should never happen (or we can keep it but change to FATAL).

Reply:
Changed to FATAL